owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt

```diff
@@ -584,19 +588,30 @@ class OCFileRepository(
 
         // 1. Remove folder content recursively
         folderContent.forEach { file ->
-            if (file.isFolder) {
-                deleteLocalFolderRecursively(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
-            } else {
-                deleteLocalFile(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
+            if (!onlyFromLocalStorage || !file.isAvailableOffline) {
```

Comment:
I can't see why you used `onlyFromLocalStorage` here, could you explain it a bit?

Reply:
`onlyFromLocalStorage` is working the same as the `removeAvailableOfflineFile` variable we had previously created. That is, when this variable is `true`, the available offline files should not be deleted and the other files will be reset. If the variable is false, the files in the database will be deleted, included av.offline.

The value of the variable in the uses of the `deleteLocalFolderRecursively` function is the same as in the previously created variable. That is, in the cases of the `catch` it is set to `false`, and in the `deleteFiles` case it is set to whatever is given to us by parameter, which could be `true` or `false`.

In conclusion, it does the same job as the `removeAvailableOfflineFile` variable.

Reply:
The condition is clearer and equivalent like this: `if (!(onlyFromLocalStorage && file.isAvailableOffline))`

Change it by this and add a clarifying comment and we're done